sftp-client.c

```diff
@@ -694,6 +699,16 @@ do_mkdir(struct sftp_conn *conn, const char *path, Attrib *a, int print_flag)
 {
 	u_int status, id;
 
+#ifdef IRONCORE
+	/*  Block use of .iron. prefix on directory names on remote server, since we use that for the files
+	 *  that store sharing info.
+	 */
+	if (iron_extension_offset(path) >= 0) {
+		error("Directory name suffix \"%s\" reserved for use on the server", IRON_SECURE_FILE_SUFFIX);
```

Comment:
Do you think it'd be a bad idea to put the original `path` into the error message? Something like:
`error("Directory name suffix \"%s\" reserved for use on the server. \"%s\ is not a permitted directory name.", IRON_SECURE_FILE_SUFFIX, name);